src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,25 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can not be null");
+        }
+
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RuntimeException("login must contain more than 6 characters");
+        }
+
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RuntimeException("Password must be longer than 6");
+        }
+
+        if (user.getAge() < 18) {
```

Comment:
The age check should be a separate condition to ensure that the message is specific to the age validation failure.